### ISSUE TITLE
Separate lines in the downloadable owncloud.log

### ISSUE THREAD

**Issue by enoch85** [owncloud/core#14268]
If you download the owncloud.log from `/index.php/settings/admin` the lines won't be separated causing the post output to be on one line instead of several different.

This should be fixed so that issue reports here on GitHub will be easier to read. 

CC @DeepDiver1975

**Comment by enoch85** [comment]
Here is a pretty good example: https://github.com/owncloud/core/issues/14021#issuecomment-73672732 

@PVince81 had to reformat my original output. This should be done automatically.

**Comment by PVince81** [comment]
I guess the original decision of using JSON for the logs was to be able to read it in external programs.

Having every log entry span over multiple lines is also not the best solution. Most log files in general are "one line per entry", so not sure.

You can actually manually auto-format it by doing this:
`cat owncloud.log | jsonlint -p -`

**Comment by PVince81** [comment]
CC @LukasReschke in case you want to add this in your bug reporting tool :smile:

**Comment by enoch85** [comment]
> You can actually manually auto-format it by doing this:
cat owncloud.log | jsonlint -p -

Problem is, I download to Windows, always.

**Comment by nickvergessen** [comment]
Just use anything else then "Editor" for coding related stuff, as mentioned everywhere on the internet.
There are even free solutions that are quite nice and enough. I used Notepad++ for coding 6 years.

But Editor of M$ is just broken.

**Comment by nickvergessen** [comment]
Or do you mean
```
{
  "reqId": "c66669e3fc541d2cdaa36ced1ca934b6",
  "remoteAddr": "46.236.021.00",
  "app": "PHP",
  "message": "Array to string conversion at /var/www/html/example.se/owncloud/lib/private/l10n/string.php#75",
  "level": 3,
  "time": "2015-02-10T00:48:34+00:00"
}
```
vs
```
{"reqId": "c66669e3fc541d2cdaa36ced1ca934b6", "remoteAddr": "46.236.021.00", "app": "PHP", "message": "Array to string conversion at /var/www/html/example.se/owncloud/lib/private/l10n/string.php#75", "level": 3, "time": "2015-02-10T00:48:34+00:00"}
```
If so, then we shouldn't really change it for usability in the console.
You however could, by changing the json_encode() call to `json_encode($data, JSON_PRETTY_PRINT)` (since php 5.4 is now required)

**Comment by enoch85** [comment]
@nickvergessen Yes, that's exactly what I mean. And btw, I use Notepad ++ and it's on one line.

**Comment by nickvergessen** [comment]
Well it's quite normal for server logs to be once line, this allows easier tracking.
So "won't fix"

**Comment by enoch85** [comment]
@nickvergessen Excuse me? You think it's easier to read one line instead of dividing it by (let's say 20) lines? The output is much harder to read. 

{
  "reqId": "c66669e3fc541d2cdaa36ced1ca934b6",
  "remoteAddr": "46.236.021.00",
  "app": "PHP",
  "message": "Array to string conversion at /var/www/html/example.se/owncloud/lib/private/l10n/string.php#75",
  "level": 3,
  "time": "2015-02-10T00:48:34+00:00"
}

**Comment by PVince81** [comment]
The owncloud.log isn't designed to be read directly. The reason it uses JSON is to be parsed by tools and log files usually have one line per entry.

I do agree that it's not human-readable, so... not sure.

**Comment by nickvergessen** [comment]
@enoch85 ? Sorry? the file is not ment to be user readable.
Just go to the admin page and copy your log from there, if you need it readable...

**Comment by enoch85** [comment]
@nickvergessen The purpose of fixing this is to get the stack trace found in the downloadable file. Ok with me to not change this as it's not affecting me - it's affecting the devs that solve the bugs. 

Lile @PVince81 said, not human readable.

Ot is it better to just attach the log file in each issue report instead of following the contrib guidlines and write it out in text instead?

**Comment by enoch85** [comment]
Sorry for typos, sent from mobile, please edit.

**Comment by enoch85** [comment]
@nickvergessen In  this example https://github.com/owncloud/core/issues/14021#issuecomment-73672732 you can see that it´s several lines.

Here are two of them:

```

  "reqId": "c66669e3fc541d2cdaa36ced1ca934b6" "remoteAddr": "46.236.021.00","app": "PHP","mess":"Array to string conversion at /var/www/html/example.se/owncloud/lib/private/l10n/string.php#75","level": 3,"time": "2015-0210T00:48:34+00:00"
  "reqId": "4694e4a550ae1bb3e31c6fa682f97c25","remoteAddr": "132.123.123.21","app": "webdav","message": "Exception: {\"Message\":\"expected filesize 1798759 got 0\",\"Code\":0,\"Trace\":\"#0 /var/www/html/example.se/owncloud/lib/private/connector/sabre/directory.php(84): OC_Connector_Sabre_File->put(Resource id #107)\n#1 /var/www/html/example.se/owncloud/3rdparty/sabre/dav/lib/Sabre/DAV/Server.php(1647): OC_Connector_Sabre_Directory->createFile('IMG_2361.JPG', Resource id #107)\n#2 /var/www/html/example.se/owncloud/3rdparty/sabre/dav/lib/Sabre/DAV/Server.php(900): Sabre\\DAV\\Server->createFile('Bilder/2012/Emm...', Resource id #107, NULL)\n#3 [internal function]: Sabre\\DAV\\Server->httpPut('Bilder/2012/Emm...')\n#4 /var/www/html/example.se/owncloud/3rdparty/sabre/dav/lib/Sabre/DAV/Server.php(474): call_user_func(Array, 'Bilder/2012/Emm...')\n#5 /var/www/html/example.se/owncloud/3rdparty/sabre/dav/lib/Sabre/DAV/Server.php(214): Sabre\\DAV\\Server->invokeMethod('PUT', 'Bilder/2012/Emm...')\n#6 /var/www/html/example.se/owncloud/apps/files/appinfo/remote.php(61): Sabre\\DAV\\Server->exec()\n#7 /var/www/html/example.se/owncloud/remote.php(54): require_once('/var/www/html/f...')\n#8 {main\",\"File\":\"/var/www/html/example.se/owncloud/lib/private/connector/sabre/file.php\",\"Line\":130}","level": 4,"time": "2015-02-09T21:45:42+00:00"

And that's what I mean by "seperate lines".
```

**Comment by nickvergessen** [comment]
That is, because we take the time to either manually edit it,
or use scripts: https://gist.github.com/nickvergessen/eb784c9b84922bdb605c

really, nothing we should fix in OC itself.

**Comment by enoch85** [comment]
Just one question: couldn't you add the script pre-downloading the owncloud.log from ` /index.php/settings/admin  `?

**Comment by MorrisJobke** [comment]
Fix in #15261
